python/pyspark/pandas/plot/matplotlib.py

```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-from distutils.version import LooseVersion
+from packaging.version import Version
```

Comment:
Oh actually we can't just add a required dependency easily. It works with `pip install` but it wouldn't work when users download Spark from the official website (as they have to manually install `packaging` dependency in their nodes) - in case of Py4J, we already contain that in our release so it works.

We should either port `packaging` into our release, or find a workaround to avoid adding the required dependency.

Reply:
I can take a look around (mid) next week too.

Reply:
Yes, currently, it should be installed manually like Numpy. 

However, it's independent from supporting Python 3.12 itself.

Can we do that porting after this?

Reply:
I can proceed to port `packaging` tomorrow too because USA doesn't have Chuseok, @HyukjinKwon . :)